src/cpptests/t_llapi.cpp

```diff
@@ -298,8 +298,88 @@ TEST_F(LLApiTest, testRanges) {
     results.insert(idstr);
   }
 
-  ASSERT_EQ(10, results.size());
-  for (char c = 'n'; c < 'x'; c++) {
+  ASSERT_EQ(11, results.size());
+  for (char c = 'n'; c <= 'x'; c++) {
+    char namebuf[64];
+    sprintf(namebuf, "doc%c", c);
+    ASSERT_NE(results.end(), results.find(namebuf));
+  }
+  RediSearch_ResultsIteratorFree(iter);
+
+  qn = RediSearch_CreateLexRangeNode(index, FIELD_NAME_1, "MarkN", "MarkX", 0, 0);
+  iter = RediSearch_GetResultsIterator(qn, index);
+  ASSERT_FALSE(NULL == iter);
+  results.clear();
+  while ((id = (const char*)RediSearch_ResultsIteratorNext(iter, index, &nid))) {
+    std::string idstr(id, nid);
+    ASSERT_EQ(results.end(), results.find(idstr));
+    results.insert(idstr);
+  }
+
+  ASSERT_EQ(9, results.size());
+  for (char c = 'o'; c <= 'w'; c++) {
+    char namebuf[64];
+    sprintf(namebuf, "doc%c", c);
+    ASSERT_NE(results.end(), results.find(namebuf));
+  }
+  RediSearch_ResultsIteratorFree(iter);
+
+  // printf("Have %lu ids in range!\n", results.size());
+  RediSearch_DropIndex(index);
+}
+
+TEST_F(LLApiTest, testRangesOnTags) {
```

Comment:
I'm sure there's a way we can refactor out some common code between those two test functions.